Doc/library/shutil.rst

```diff
@@ -433,23 +433,44 @@ Directory and files operations
    When no *path* is specified, the results of :func:`os.environ` are used,
    returning either the "PATH" value or a fallback of :attr:`os.defpath`.
 
-   On Windows, the current directory is always prepended to the *path* whether
-   or not you use the default or provide your own, which is the behavior the
-   command shell uses when finding executables.  Additionally, when finding the
-   *cmd* in the *path*, the ``PATHEXT`` environment variable is checked.  For
-   example, if you call ``shutil.which("python")``, :func:`which` will search
-   ``PATHEXT`` to know that it should look for ``python.exe`` within the *path*
-   directories.  For example, on Windows::
+   On Windows, the current directory is prepended to the *path* if
+   the *mode* does not include ``os.X_OK``. When the *mode* does include
+   ``os.X_OK``, the Windows API ``NeedCurrentDirectoryForExePathW`` will be
+   consulted to determine if the current directory should be prepended to
+   *path*. To avoid consulting the current working directory for executables,
+   and thereby, match the behavior of non-Windows shells: set the environment
+   variable ``NoDefaultCurrentDirectoryInExePath``.
+
+   Also on Windows, the ``PATHEXT`` variable is used to resolve commands
+   that may not already include an extension. For example, if you call
+   ``shutil.which("python")``, :func:`which` will search ``PATHEXT``
+   to know that it should look for ``python.exe`` within the *path*
+   directories. For example, on Windows::
 
       >>> shutil.which("python")
       'C:\\Python33\\python.EXE'
 
+   Similar ``PATHEXT`` logic is also applied when a full path to a *cmd* is
```

Comment:
"full path" or "fully qualified path" means an absolute path, but the logic here applies to any qualified path, or stated simply: "... also applied when \*cmd\* is a path that contains a directory component::"